### ISSUE TITLE
Okta application grants

### ISSUE THREAD

**Issue by r0mant** [gravitational/teleport#20972]
To solve the problem with granting a long-term access to users with periodic audits and tracking of "who gave access to whom and why and when" we want to introduce a concept of "access group" as described in this (still WIP) [RFD](https://github.com/gravitational/teleport.e/pull/717).

**Comment by klizhentas** [comment]
Here is an interesting use-case connecting some dots. Follow my stream of thought. I have been researching ZScaler ZPA. One of their benefits is that internal apps will not going to be visible unless a user logs in on the client. This works similarly to Teleport, but via DNS and VPNs. Now, usually this is not enough, as users also need to login in the app. This is where Okta comes in with their SSO.
Today, I requested access to Posthog. I created a PR https://github.com/gravitational/cloud-terraform/pull/1488/files. This is implemented by Travis and their team with Spacelift parsing CSV and adding me to Okta groups.
Once the PR got approved, I went to reporting.platform.teleport.sh and was able to visit Posthog app hosted on our infra. Now, Posthog asked me to login (again) and I used our configured Okta SSO.
So here is how our App grants + IDP make this process trivial:
User will create an [application grant](https://github.com/gravitational/teleport.e/pull/717/files) pull request in Terraform. Once PR is merged, Spacelift will apply this PR and user will be able to access PostHog app. PostHog will be configured to trust Teleport as IDP, not Okta. Okta upstream IDP will only be used to store emails and the remaining grant logic will be completely handled by Teleport.
This model not only has all the advantages of ZPA, but also all the advantages of Okta.
What I like about this is that Okta is not a critical component of this exchange, so users may choose to not rely on on it as the source of their identity, but this is not a requirement for them to benefit from this feature.
